File: metadata/api-tests/infraestructure_support.py
import requests
import json

def obtener_vias_osm(lat_min, lon_min, lat_max, lon_max):
    """
    Consulta OpenStreetMap vía Overpass API para obtener vías en el bounding box con tags.
    Retorna un GeoJSON-like dict con las vías y sus propiedades.
    """
    overpass_url = "http://overpass-api.de/api/interpreter"
    # Query OSM Overpass QL para formas de vías (way) con tag highway dentro del bbox
    query = f"""
    [out:json][timeout:25];
    (
      way["highway"]({lat_min},{lon_min},{lat_max},{lon_max});
    );
    out body;
    >;
    out skel qt;
    """
    response = requests.get(overpass_url, params={'data': query})
    data = response.json()
    return data

if __name__ == "__main__":
    # Ejemplo: bounding box aproximado Biobío (puedes ajustarlo)
    lat_min, lon_min = -37.5, -73.5
    lat_max, lon_max = -36.5, -72.5
    vias = obtener_vias_osm(lat_min, lon_min, lat_max, lon_max)
    print("Número de vías encontradas:", len(vias.get('elements', [])))
    # Mostrar algunas vías con tags
    for e in vias.get('elements', [])[:5]:
        print(e.get('id'), e.get('tags'))

File: metadata/infraestructure_support.js
const axios = require('axios');

async function obtenerViasOSM(latMin, lonMin, latMax, lonMax) {
    /**
     * Consulta OpenStreetMap vía Overpass API para obtener vías en el bounding box con tags.
     * Retorna un objeto GeoJSON-like con las vías y sus propiedades.
     */
    const overpassUrl = "https://overpass-api.de/api/interpreter";
    
    // Query optimizada: solo vías principales, con centro para geolocalización
    const query = `
    [out:json][timeout:180][bbox:${latMin},${lonMin},${latMax},${lonMax}];
    way["highway"~"motorway|trunk|primary|secondary|tertiary"];
    out tags center;
    `;

    try {
        console.log("Consultando Overpass API para vías...");
        const response = await axios.post(
            overpassUrl,
            `data=${encodeURIComponent(query)}`,
            {
                headers: {
                    'Content-Type': 'application/x-www-form-urlencoded'
                },
                timeout: 200000 // 200 segundos
            }
        );

        console.log("✓ Datos de vías obtenidos exitosamente");
        return response.data;
    } catch (error) {
        if (error.code === 'ECONNABORTED' || error.response?.status === 504) {
            console.error("Timeout: El área es demasiado grande. Intenta con un bounding box más pequeño.");
        } else {
            console.error("Error al obtener vías OSM:", error.message);
        }
        return null;
    }
}

async function main() {
    // Área muy pequeña del centro de Concepción (~2km x 2km)
    const latMin = -36.83, lonMin = -73.06;
    const latMax = -36.81, lonMax = -73.04;
    
    console.log(`Consultando vías en el área: (${latMin},${lonMin}) a (${latMax},${lonMax})`);
    
    const vias = await obtenerViasOSM(latMin, lonMin, latMax, lonMax);
    
    if (vias) {
        const elementos = vias.elements || [];
        console.log("Número de vías encontradas:", elementos.length);
        
        // Mostrar algunas vías con tags
        elementos.slice(0, 5).forEach(e => {
            console.log(e.id, e.tags);
        });
    }
}

// Exportar la función
module.exports = { obtenerViasOSM };

// Si se ejecuta directamente
if (require.main === module) {
    main().catch(console.error);
}
